llvm/lib/SYCLLowerIR/LowerESIMD.cpp

```diff
@@ -1387,6 +1387,12 @@ PreservedAnalyses SYCLLowerESIMDPass::run(Module &M,
 
 size_t SYCLLowerESIMDPass::runOnFunction(Function &F,
                                          SmallPtrSet<Type *, 4> &GVTS) {
+  // ITT annotation functions are an exclusion from ESIMD rules regarding loads
```

Comment:
I think the approach to optimizing compilation time by filtering out __itt function early in an LLVM pass should be "all or none". Is only ESIMD lowering pass benefits from this early bail-out? If there are other passes (which I think is likely) then it would be logical to apply this optimization to all of them.

Or LowerESIMD crashes w/o this change?

Reply:
Crashes in the ESIMD pass are the reason for this change. That said, I could investigate if I can quickly make `__itt` functions skippable by most/all of the passes.

Reply:
Thanks! NIT: If that skipping can't be done, then I think it'd be good to extend the above comment with why these functions crashe ESIMD.

Reply:
Per sync with @MrSidims, I've disabled the annotation of ESIMD kernels entirely. If the ITT annotations are needed for GPU-specific code at some point, a separate re-design of the instrumentation feature will be required anyway.